src/runtime/methodbinder.cs

```diff
@@ -98,12 +112,47 @@ internal static MethodInfo MatchParameters(MethodInfo[] mi, Type[] tp)
             return null;
         }
 
+        internal static IEnumerable<MethodInfo> MatchParamertersMethods(IEnumerable<MethodInfo> mi, Type[] tp)
+        {
+            if (tp == null)
+            {
+                yield break;
```

Comment:
I think this should throw instead, and the caller should explicitly handle `null` case.